src/app/file/bmp_format.cpp

```diff
@@ -151,26 +243,74 @@ static int read_bmfileheader(FILE *f, BITMAPFILEHEADER *fileheader)
 /* read_win_bminfoheader:
  *  Reads information from a BMP file header.
  */
-static int read_win_bminfoheader(FILE *f, BITMAPINFOHEADER *infoheader)
+static int read_win_bminfoheader(FILE *f, BITMAPINFOHEADER *infoheader, int biSize)
 {
   WINBMPINFOHEADER win_infoheader;
 
+  if (biSize != 16 &&
+      biSize != 24 &&
+      biSize != WININFOHEADERSIZE &&
+      biSize != BV2INFOHEADERSIZE &&
+      biSize != BV3INFOHEADERSIZE &&
+      biSize != 60 &&
+      biSize != 64) return -1;
+  
+  bool isRGBMasks = false, isAlphaMask = false;
+
   win_infoheader.biWidth = fgetl(f);
   win_infoheader.biHeight = fgetl(f);
   win_infoheader.biPlanes = fgetw(f);
-  win_infoheader.biBitCount = fgetw(f);
-  win_infoheader.biCompression = fgetl(f);
-  win_infoheader.biSizeImage = fgetl(f);
-  win_infoheader.biXPelsPerMeter = fgetl(f);
-  win_infoheader.biYPelsPerMeter = fgetl(f);
-  win_infoheader.biClrUsed = fgetl(f);
-  win_infoheader.biClrImportant = fgetl(f);
+  win_infoheader.biBitCount = fgetw(f);// = 16 bytes
+
+  win_infoheader.redMask = 0;
+  win_infoheader.greenMask = 0;
+  win_infoheader.blueMask = 0;
+  win_infoheader.alphaMask = 0;
+  if (biSize == 16)
+    win_infoheader.biCompression = BI_RGB;
+  else { // biSize >= 24
+    win_infoheader.biCompression = fgetl(f);
```

Comment:
You can convert these kind of comments in asserts:
```suggestion
  else {
    ASSERT(biSize >= 24);
    win_infoheader.biCompression = fgetl(f);
```